From 9165b11f689f18d160ad059ab3c7a43b2bb39533 Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Thu, 25 Feb 2016 11:17:18 +0800
Subject: [PATCH 21/23] utime.h: fix timezone issue in round_to_* funcs.

gmtime_r converts local time to UTC, however mktime only takes an
argument as local time. Use localtime_r instead of gmtime_r will fix.

Fixes: #14862

Reported-by: isyippee <yippee_liu@163.com>
Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
(cherry picked from commit c914f28d6e79726b84261884c37b89d7c6b34f21)
---
 src/include/utime.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/include/utime.h b/src/include/utime.h
index 3108ecd..1467a6b 100644
--- a/src/include/utime.h
+++ b/src/include/utime.h
@@ -110,7 +110,7 @@ public:
   utime_t round_to_minute() {
     struct tm bdt;
     time_t tt = sec();
-    gmtime_r(&tt, &bdt);
+    localtime_r(&tt, &bdt);
     bdt.tm_sec = 0;
     tt = mktime(&bdt);
     return utime_t(tt, 0);
@@ -119,7 +119,7 @@ public:
   utime_t round_to_hour() {
     struct tm bdt;
     time_t tt = sec();
-    gmtime_r(&tt, &bdt);
+    localtime_r(&tt, &bdt);
     bdt.tm_sec = 0;
     bdt.tm_min = 0;
     tt = mktime(&bdt);
-- 
2.8.0.rc3

